planner/optimize.go

```diff
@@ -279,6 +279,18 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 					sessVars.StmtCtx.AppendWarning(err)
 				}
 			}
+
+			// Override the resource group if necessary
+			// TODO: we didn't check the existence of the hinted resource group now to save the cost per query
+			if curStmtHints.HasResourceGroup {
```

Comment:
I think you can just move the code of L185~L192 after this `if useBinding {` to avoiding duplicate this code block twice

Reply:
Done. PTAL, thx~